packages/binding/src/definition/definition.ts

```diff
@@ -1,251 +1,167 @@
+        notAllowedElements: [BindingInfoName.parts, BindingInfoName.value],
+      });
+      break;
+    case BindingInfoName.value: {
+      types.push({
+        kind: type,
         dependents: [],
-        notAllowedElements: [],
-      },
-    ],
-    description: {
-      text: "Function to convert model data into a property value",
-      visibility: "Public",
-    },
-  },
-  {
-    name: "useRawValues",
-    type: [
-      {
-        kind: "boolean",
+        notAllowedElements: [BindingInfoName.parts, BindingInfoName.path],
+      });
+      break;
+    }
+    case BindingInfoName.parts: {
+      types.push({
+        kind: TypeKind.object,
         dependents: [],
-        notAllowedElements: [],
-      },
-    ],
-    description: {
-      text: `Whether the parameters to the formatter function should be passed as raw values. In this case the specified types for the binding parts are not used and the values are not formatted.
-**Note**: use this flag only when using multiple bindings. If you use only one binding and want raw values then simply don't specify a type for that binding`,
-      visibility: "Public",
-    },
-  },
-  {
-    name: "useInternalValues",
-    type: [
-      {
-        kind: "boolean",
+        notAllowedElements: [BindingInfoName.value, BindingInfoName.path],
+        collection: true,
+      });
+      types.push({
+        kind: TypeKind.string,
         dependents: [],
         notAllowedElements: [],
-      },
-    ],
-    description: {
-      text: `Whether the parameters to the formatter function should be passed as the related JavaScript primitive values. In this case the values of the model are parsed by the model format of the specified types from the binding parts.
-**Note**: use this flag only when using multiple bindings.`,
-      visibility: "Public",
-    },
-  },
-  {
-    name: "type",
-    type: [
-      {
-        kind: "object",
+        collection: true,
+      });
+      break;
+    }
+    case BindingInfoName.suspended:
+    case BindingInfoName.useRawValues:
+    case BindingInfoName.useInternalValues: {
+      types.push({
+        kind: type,
         dependents: [],
         notAllowedElements: [],
-      },
-      {
-        kind: "string",
+        default: {
+          values: [true, false],
+          fixed: true,
+        },
+      });
+      break;
+    }
+    case BindingInfoName.mode: {
+      types.push({
+        kind: TypeKind.string,
         dependents: [],
         notAllowedElements: [],
-      },
-    ],
-    description: {
-      text: `A type object or the name of a type class to create such a type object; the type will be used for converting model data to a property value (aka "formatting") and vice versa (in binding mode TwoWay, aka "parsing")`,
-      visibility: "Public",
-    },
-  },
-  {
-    name: "targetType",
-    type: [
-      {
-        kind: "string",
+        default: {
+          values: [
+            "sap.ui.model.BindingMode.Default",
```

Comment:
We have this information in the semantic model, I don't think we should hard code it here.